### ISSUE TITLE
StarTeam v16

### ISSUE THREAD

**Issue by HTA-IT** [planestraveler/git-starteam#62]
Do you know if this works against StarTeam SDK v16?

**Comment by planestraveler** [comment]
Sorry for the long delay but here is a question for you.

I have never come across the v16 starteam SDK, does it contains a similar file to `starteam110.bridgeTo.starteam125.jar` which could be used to bridge the gap between the old version of the API and the newest version ?

**Comment by millerdt2** [comment]
Hello, I'm currently using SDK v15 and it has starteam110.bridgeTo.starteam150.jar.  Do you think I can use your migrator with v15.  Sorry but we dont have any java developers to work on this, we are strictly .net.  Thanks.

**Comment by planestraveler** [comment]
Hello,

Yes it shall be sufficient to have this jar in the classpath to execute the conversion. Here is an example on how the command line should present to you to launch the main class.
```
java -cp jargs.jar:starteam150.jar:starteam110.bridgeTo.starteam150.jar:syncronizer.jar org.sync.MainEntry --help
```
Make sure the paths are all set correctly for each jarfile.

Have a good day.

**Comment by millerdt2** [comment]
Thanks for the response.  I updated my classpath to all of the jar files except syncronizer.jar because I don't know where it is located.  When running the above command I get the following: 

C:\windows\system32>java -cp jargs.jar:starteam150.jar:starteam110.bridgeTo.star
team150.jar:syncronizer.jar org.sync.MainEntry --help
Error: Could not find or load main class org.sync.MainEntry
Caused by: java.lang.ClassNotFoundException: org.sync.MainEntry

**Comment by planestraveler** [comment]
Yeah the most critical of all jar files is actually `synchronizer.jar`. If you did build from sources using ant, it should be located in the `bin` folder at the root of the project. For example, mise is located here `/home/xxxxxxxxxx/java/git-starteam/bin/syncronizer.jar`

**Comment by millerdt2** [comment]
I'm still having problems building from source.  I discovered your tool won't work with Jdk9. Now that I've downgraded to jdk8 I'm getting the following errors:
[migrationtoolerror.txt](https://github.com/planestraveler/git-starteam/files/1598602/migrationtoolerror.txt)

Thanks.

**Comment by jbjonesjr** [comment]
@millerdt2 looks like your are missing [jargs](http://jargs.sourceforge.net/) from your classpath. Available in jar from maven central at: https://search.maven.org/remotecontent?filepath=com/talanlabs/jargs/1.0.0/jargs-1.0.0.jar

**Comment by millerdt2** [comment]
Thanks, i was able to build from source.  I now have fake-starteam.jar and syncronizer.jar and I've added them to the classpath.  However, I'm still getting the following error:
Error: Could not find or load main class org.sync.MainEntry
Caused by: java.lang.ClassNotFoundException: org.sync.MainEntry

**Comment by planestraveler** [comment]
@millerdt2 please remove the fake-starteam.jar from the classpath, this one will conflict with the actual starteam sdk. This jar file only exist for mocking starteam operations for testing, hence the name `fake-starteam`.

Strange thing that the java process cannot find the class in the classpath. Can we see what is the value you pass as parameters to `-cp` command line argument ?

**Comment by millerdt2** [comment]
I think I'm almost there.  I now have a zipped file (StarteamFileInfo.gz) that I can't unzip but can look at its contents.
![image](https://user-images.githubusercontent.com/28312369/34996215-e8dc399e-faa6-11e7-9697-8adf50c841cf.png)
[StarteamFileInfo.gz](https://github.com/planestraveler/git-starteam/files/1635691/StarteamFileInfo.gz)

**Comment by millerdt2** [comment]
Fake-starteam.jar has already been removed.  When i use -cp i get an error: 
C:\git-migration>java -cp jargs.jar:starteam150.jar:syncronizer.jar org.sync.MainEntry -h
10.0.4.49 -P 49001 -v AAMS -p AAMS -H master -U appmillerdt -c
Error: Could not find or load main class org.sync.MainEntry
Caused by: java.lang.ClassNotFoundException: org.sync.MainEntry

However, when I don't use -cp I get some results:

c:\git-migration>java org.sync.MainEntry -h 10.0.4.49 -P 49001 -v AAMS -p AAMS -c -W repo1
 -H master -R
Username:appmillerdt
Password:
2018-01-10 12:50:19 Failed to checkout Documentation/Deployment Plan/AAMS 5.00 - Deploymen
t Plan.docx: Missing archive D:\StarTeam-Repository\AP\APHIVE\Archive\a7\7\a778abf53ece76b
bbb8419845f767543
2018-01-10 12:50:19 Failed to checkout Documentation/Deployment Plan/AAMS 5.00 - Deploymen
t Plan_v1.1.docx: Missing archive D:\StarTeam-Repository\AP\APHIVE\Archive\a7\7\a778abf53e
ce76bbbb8419845f767543
2018-01-10 12:50:19 Failed to checkout Source/AAMSWeb/Business/ActivityRepository.cs: Miss
ing archive D:\StarTeam-Repository\AP\APHIVE\Archive\a7\7\a778abf53ece76bbbb8419845f767543

2018-01-10 12:50:19 Failed to checkout Source/AAMSWeb/connection.config: Missing archive D
:\StarTeam-Repository\AP\APHIVE\Archive\a7\7\a778abf53ece76bbbb8419845f767543
2018-01-10 12:50:19 Failed to checkout Source/AAMSWeb/AAMSWeb.csproj: Missing archive D:\S
tarTeam-Repository\AP\APHIVE\Archive\a7\7\a778abf53ece76bbbb8419845f767543
2018-01-10 12:50:19 Failed to checkout Source/AAMSWeb/app.config: Missing archive D:\StarT
eam-Repository\AP\APHIVE\Archive\a7\7\a778abf53ece76bbbb8419845f767543
2018-01-10 12:50:19 Failed to checkout Source/Shared Lib/Other/Oracle.DataAccess.dll: Miss
ing archive D:\StarTeam-Repository\AP\APHIVE\Archive\a7\7\a778abf53ece76bbbb8419845f767543

2018-01-10 12:50:19 There was no new revision in the starteam view.
2018-01-10 12:50:19 All the files in the repository are at their latest version
2018-01-10 12:50:20 fast-import:C:\Program Files\Git\mingw64\libexec\git-core\git-fast-imp
ort.exe statistics:
2018-01-10 12:50:20 fast-import:----------------------------------------------------------
-----------
2018-01-10 12:50:20 fast-import:Alloc'd objects:       5000
2018-01-10 12:50:20 fast-import:Total objects:            0 (         0 duplicates
          )
2018-01-10 12:50:20 fast-import:      blobs  :            0 (         0 duplicates
  0 deltas of          0 attempts)
2018-01-10 12:50:20 fast-import:      trees  :            0 (         0 duplicates
  0 deltas of          0 attempts)
2018-01-10 12:50:20 fast-import:      commits:            0 (         0 duplicates
  0 deltas of          0 attempts)
2018-01-10 12:50:20 fast-import:      tags   :            0 (         0 duplicates
  0 deltas of          0 attempts)
2018-01-10 12:50:20 fast-import:Total branches:           0 (         0 loads     )
2018-01-10 12:50:20 fast-import:      marks:           1024 (         0 unique    )
2018-01-10 12:50:20 fast-import:      atoms:              0
2018-01-10 12:50:20 fast-import:Memory total:          2282 KiB
2018-01-10 12:50:20 fast-import:       pools:          2048 KiB
2018-01-10 12:50:20 fast-import:     objects:           234 KiB
2018-01-10 12:50:20 fast-import:----------------------------------------------------------
-----------
2018-01-10 12:50:20 fast-import:pack_report: getpagesize()            =      65536
2018-01-10 12:50:20 fast-import:pack_report: core.packedGitWindowSize = 1073741824
2018-01-10 12:50:20 fast-import:pack_report: core.packedGitLimit      = 8589934592
2018-01-10 12:50:20 fast-import:pack_report: pack_used_ctr            =          0
2018-01-10 12:50:20 fast-import:pack_report: pack_mmap_calls          =          0
2018-01-10 12:50:20 fast-import:pack_report: pack_open_windows        =          0 /
    0
2018-01-10 12:50:20 fast-import:pack_report: pack_mapped              =          0 /
    0
2018-01-10 12:50:20 fast-import:----------------------------------------------------------
-----------
2018-01-10 12:50:20 fast-import:

**Comment by planestraveler** [comment]
@millerdt2 Seem to me you got it working, the command line you provided must have imported the repository into the git-migration folder. You must have properly set your classpath environment variable. 👍  You now have your setup to go and convert your full buisness to git 😄 .

**Comment by millerdt2** [comment]
The problem is I don't see the repository in the git-migration folder.  What am I missing?

**Comment by vectorphresh** [comment]
Could it be your classpath separator? Should be semicolon on windows.
https://howtodoinjava.com/java/basics/java-classpath/
